linux/LibSensors.c

```diff
@@ -105,99 +99,146 @@ void LibSensors_cleanup(void) {
 #endif /* BUILD_STATIC */
 }
 
+static int tempDriverPriority(const sensors_chip_name* chip) {
+   static const struct TempDriverDefs {
+      const char* prefix;
+      int priority;
+   } tempDrivers[] =  {
+      { "coretemp",    0 },
+      { "via_cputemp", 0 },
+      { "cpu_thermal", 0 },
+      { "k10temp",     0 },
+      { "zenpower",    0 },
+      /* Low priority drivers */
+      { "acpitz",      1 },
+   };
+
+   for (size_t i = 0; i < ARRAYSIZE(tempDrivers); i++)
+      if (String_eq(chip->prefix, tempDrivers[i].prefix))
+         return tempDrivers[i].priority;
+
+   return -1;
+}
+
 void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int cpuCount) {
-   for (unsigned int i = 0; i <= cpuCount; i++)
-      cpus[i].temperature = NAN;
+   double data[cpuCount + 1];
+#ifdef __clang_analyzer__
+   memset(data, -1, ARRAYSIZE(data)); /* please clang analyzer */
+#endif
+   for (size_t i = 0; i < ARRAYSIZE(data); i++)
+      data[i] = NAN;
 
 #ifndef BUILD_STATIC
    if (!dlopenHandle)
-      return;
+      goto out;
 #endif /* !BUILD_STATIC */
 
    unsigned int coreTempCount = 0;
+   int topPriority = 99;
 
    int n = 0;
-   for (const sensors_chip_name *chip = sym_sensors_get_detected_chips(NULL, &n); chip; chip = sym_sensors_get_detected_chips(NULL, &n)) {
-      char buffer[32];
-      sym_sensors_snprintf_chip_name(buffer, sizeof(buffer), chip);
-      if (!String_startsWith(buffer, "coretemp") &&
-          !String_startsWith(buffer, "cpu_thermal") &&
-          !String_startsWith(buffer, "k10temp") &&
-          !String_startsWith(buffer, "zenpower"))
+   for (const sensors_chip_name* chip = sym_sensors_get_detected_chips(NULL, &n); chip; chip = sym_sensors_get_detected_chips(NULL, &n)) {
+      const int priority = tempDriverPriority(chip);
+      if (priority < 0)
+         continue;
+
+      if (priority > topPriority)
          continue;
 
+      topPriority = priority;
```

Comment:
may still mix values from different priorities depending on reported order of sensors.

Reply:
Should be fixed now